.gitbook/contents/developer/features/custom-fee-token.md

```diff
@@ -0,0 +1,115 @@
+# Custom Fee Token
+
+## Overview
+
+Custom gas token allows for an L1-native ERC20 token to collateralize and act as the gas token on L2. This implementation is based on [Optimism's Custom Fee Token](https://specs.optimism.io/protocol/granite/custom-gas-token.html) and enhances its functionality by enabling the bridging of L1 native tokens to L2 as ERC20 tokens. 
+
+**The codebase is located in the [custom-fee-token](https://github.com/bobanetwork/boba/tree/custom-fee-token) branch.**
+
+## Configuring the Gas Paying Token and L2 ETH Token
+
+The gas-paying token and the L2 ETH token are set within the L1 `SystemConfig` smart contract. The gas-paying token is set during initialization and cannot be modified by the `SystemConfig` bytecode. The L2 ETH token is set during initialization and can be updated via the `setL2ETHToken` function if the L2 ETH token address is `address(0)`. Since the `SystemConfig` is proxied, it is always possible to modify the storage slot that holds the gas-paying token address and the L2 ETH token address directly during an upgrade.
+
+If the address in the `GAS_PAYING_TOKEN_SLOT` slot for `SystemConfig` is `address(0)`, the system is configured to use `ether` as the gas paying token, and the getter for the token returns `ETHER_TOKEN_ADDRESS`. If the address in the `GAS_PAYING_TOKEN_SLOT` slot for `SystemConfig` is not `address(0)`, the system is configured to use a custom gas paying token, and the getter returns the address in the slot.
+
+If the gas paying token is configured and the address in the `L2_ETH_TOKEN_SLOT` slot for `SymstemConfig` is `address(0)`, the system is configured to block the ETH deposit and withdrawal. If the address in the `L2_ETH_TOKEN_SLOT` slot for `SystemConfig` is not `address(0)`, the system is configured to allow the ETH deposit and withdrawal.
```

Comment:
If the gas paying token is configured and L2_ETH_TOKEN_SLOT is addr(0), then the system should be configured to allow ETH deposit and withdrawal? (or) can we mention where "is configured to block the ETH deposit and withdrawal"

Reply:
If the `L2_ETH_TOKEN_SLOT` is not set, the smart contracts will block ETH deposits and withdrawals.

Reply:
ok, got it!